cantools/database/utils.py

```diff
@@ -78,6 +79,10 @@ def _encode_fields(fields: Sequence[Union["Signal", "Data"]],
             unpacked[field.name] = _transform(value)  # type: ignore[operator]
             continue
 
+        if isinstance(value, Decimal) and isinstance(field, Signal):
+            unpacked[field.name] = (value - field.decimal.offset) / field.decimal.scale  # type: ignore[operator]
```

Comment:
You shouldn't ignore mypy here. `offset` and `scale` can be `None`

Reply:
Hi, thanks for the comment. 

If the user wants to encode a Decimal into a signal, the code **should** break when the decimal offset and scale are not available. If they are available, my change implements the correct calculation.

Alternatively, we can fall back to using float offset and scale, but this defeats the purpose of the Decimal - making sure that the physical representation of the signal is not affected by rounding errors.

Reply:
> If the user wants to encode a Decimal into a signal, the code should break when the decimal offset and scale are not available. If they are available, my change implements the correct calculation.

I basically agree with both of you: I think that this case should get checked and if it applies an exception should be raised explicitly...

Reply:
Sure, I will implement explicit check for this case (and make sure there are no conflicts with master) next week.

Reply:
Can bitstruct encode `decimal.Decimal` instances? Do we need to cast to int or float here?
We should add a test for this.